libs/exponential-histogram/src/main/java/org/elasticsearch/exponentialhistogram/ExponentialHistogramMerger.java

```diff
@@ -69,7 +69,8 @@ private ExponentialHistogramMerger(int bucketLimit, ExponentialHistogramCircuitB
 
     // Only intended for testing, using this in production means an unnecessary reduction of precision
     private ExponentialHistogramMerger(int bucketLimit, int maxScale, ExponentialHistogramCircuitBreaker circuitBreaker) {
-        this.bucketLimit = bucketLimit;
+        // We need at least four buckets to represent any possible distribution
+        this.bucketLimit = Math.max(4, bucketLimit);
```

Comment:
Hm, not sure how I feel about silently ignoring the requested bucket limit. This sounds like a classical case for an `IllegalArgumentException`.